### PR TITLE
Fix JCasC lifecycle bug

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
@timja @v1v @kuisathaverat please review

File: src/test/java/io/jenkins/plugins/opentelemetry/LifecycleDebugger.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
```
Comment (v1v) [inline src/test/java/io/jenkins/plugins/opentelemetry/LifecycleDebugger.java:1]: A test class?

File: src/test/java/io/jenkins/plugins/opentelemetry/BaseIntegrationTest.java [hunk 2]
```diff
@@ -54,6 +56,7 @@
 import java.util.stream.Collectors;
 
 import static com.google.common.base.Verify.verify;
+import static java.util.Optional.*;
```
Comment (v1v) [inline src/test/java/io/jenkins/plugins/opentelemetry/BaseIntegrationTest.java:59]: Nit: include all the classes?

**Review by v1v — COMMENTED** [review]
Thanks Cyrille, LGTM, I'll test this out tomorrow morning.

**Review by kuisathaverat — APPROVED** [review]

**Review by timja — APPROVED** [review]
I haven't tested but can you confirm that you've re-configured it with JCasC after an initial init and reconfiguration happens fine?

e.g. changing the url to a new port for the collector?

**Comment by v1v** [comment]
I've just tried this particular version locally from scratch and the `initializeOtlp` was triggered and configured just before all the Jobs were loaded, so It seems to work as expected, there is no need to save to load the configuration again.

```
Warning: (script, line 24) triggers is deprecated
2022-05-23 09:28:08.191+0000 [id=40]	INFO	j.j.plugin.JenkinsJobManagement#createOrUpdateConfig: createOrUpdateConfig for generate-load
2022-05-23 09:28:08.228+0000 [id=35]	INFO	jenkins.InitReactorRunner$1#onAttained: System config adapted
2022-05-23 09:28:08.915+0000 [id=77]	INFO	o.j.p.g.ApiRateLimitChecker$RateLimitCheckerAdapter#checkRateLimit: LocalChecker for rate limit was not set for this thread. Configured using system settings with API URL 'https://api.github.com'.
2022-05-23 09:28:12.246+0000 [id=77]	INFO	j.b.MultiBranchProject$BranchIndexing#run: antifraud/antifraud #20220523.092807 branch indexing action completed: SUCCESS in 5.1 sec
2022-05-23 09:28:13.758+0000 [id=35]	INFO	i.j.p.o.OpenTelemetrySdkProvider#initializeOtlp: OpenTelemetry SDK initialized: SDK [config: otel.traces.exporter=otlp, otel.metrics.exporter=otlp, otel.logs.exporter=otlp, otel.exporter.otlp.endpoint=http://otel-collector:4317, resource: service.name=jenkins, service.namespace=jenkins, service.version=2.332.3]
2022-05-23 09:28:13.877+0000 [id=45]	INFO	jenkins.InitReactorRunner$1#onAttained: Loaded all jobs
2022-05-23 09:28:13.881+0000 [id=43]	INFO	jenkins.InitReactorRunner$1#onAttained: Configuration for all jobs updated
...
```

Although, https://github.com/jenkinsci/opentelemetry-plugin/issues/437 is still an issue

File: src/test/java/io/jenkins/plugins/opentelemetry/LifecycleDebugger.java (hunk 1 above)
Comment (cyrille-leclerc) [reply src/test/java/io/jenkins/plugins/opentelemetry/LifecycleDebugger.java:1]: yes to understand the lifecycle. Note that there is a bug in `io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule`, it's not the same lifecycle, the JCasC configuration happens after `hudson.init.InitMilestone#COMPLETED` when it happens on `hudson.init.InitMilestone#SYSTEM_CONFIG_ADAPTED` in real life.

**Comment by cyrille-leclerc** [comment]
Available in https://github.com/jenkinsci/opentelemetry-plugin/releases/tag/opentelemetry-2.7.1-rc1

**Pull request merged**